src/lib/operators/table_scan/column_vs_value_table_scan_impl.cpp

```diff
@@ -65,33 +65,34 @@ void ColumnVsValueTableScanImpl::_scan_segment(const BaseSegment& segment, const
-  /*
-   * ValueID value_id; // left value id
-   * Variant value; // right value
+  /**
+   * ValueID search_vid; // left value id
+   * AlLTypeVariant search_vid_value; // dict.value_by_value_id(search_vid)
+   * Variant _value; // right value
    *
    * A ValueID value_id from the attribute vector is included in the result iff
    *
-   * Operator           |  Condition
-   * value_id == value  |  dict.value_by_value_id(dict.lower_bound(value)) == value && value_id == dict.lower_bound(value)
-   * value_id != value  |  dict.value_by_value_id(dict.lower_bound(value)) != value || value_id != dict.lower_bound(value)
-   * value_id <  value  |  value_id < dict.lower_bound(value)
-   * value_id <= value  |  value_id < dict.upper_bound(value)
-   * value_id >  value  |  value_id >= dict.upper_bound(value)
-   * value_id >= value  |  value_id >= dict.lower_bound(value)
+   * Operator          |  Condition
+   * column == _value  |  dict.value_by_value_id(dict.lower_bound(_value)) == _value && value_id == dict.lower_bound(_value)
+   * column != _value  |  dict.value_by_value_id(dict.lower_bound(value)) != _value || value_id != dict.lower_bound(_value)
+   * column <  _value  |  value_id < dict.lower_bound(_value)
+   * column <= _value  |  value_id < dict.upper_bound(_value)
+   * column >  _value  |  value_id >= dict.upper_bound(_value)
+   * column >= _value  |  value_id >= dict.lower_bound(_value)
    */
 
   const auto search_value_id = _get_search_value_id(segment);
 
   /**
    * Early Outs
    *
-   * Operator          | All                                   | None
-   * value_id == value | !None && unique_values_count == 1     | search_vid == dict.upper_bound(value)
-   * value_id != value | search_vid == dict.upper_bound(value) | !All && unique_values_count == 1
-   * value_id <  value | search_vid == INVALID_VALUE_ID        | search_vid == 0
-   * value_id <= value | search_vid == INVALID_VALUE_ID        | search_vid == 0
-   * value_id >  value | search_vid == 0                       | search_vid == INVALID_VALUE_ID
-   * value_id >= value | search_vid == 0                       | search_vid == INVALID_VALUE_ID
+   * Operator         | All                                                     | None
+   * column == _value | search_vid_value == _value && unique_values_count == 1  | search_vid_value != _value
+   * column != _value | search_vid_value != _value                              | search_vid_value == _value && unique_values_count == 1
+   * column <  _value | search_vid == INVALID_VALUE_ID                          | search_vid == 0
+   * column <= _value | search_vid == INVALID_VALUE_ID                          | search_vid == 0
+   * column >  _value | search_vid == 0                                         | search_vid == INVALID_VALUE_ID
+   * column >= _value | search_vid == 0                                         | search_vid == INVALID_VALUE_ID
```

Comment:
Do we actually test these shortcuts?

Reply:
We test that value-out-of-dictionary scans work:
ScanOnCompressedSegmentsValueLessThanMinDictionaryValue
ScanOnCompressedSegmentsValueGreaterThanMaxDictionaryValue
ScanOnCompressedSegmentsAroundBounds

Reply:
I think this is sufficient to say that this works? I mean, we do not actually test that chunks are "skipped", but how would you, without mocking?